lib/rubocop/cop/rails/migration_file_name.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop makes sure that each migration file defines a migration class
+      # whose name matches the file name.
+      # (i.e. `20220224111111_create_users.rb` should define `CreateUsers` class.)
+      #
+      # @example
+      #   # db/migrate/20220224111111_create_users.rb
+      #
+      #   # bad
+      #
+      #   class SellBooks < ActiveRecord::Migration[7.0]
+      #   end
+      #
+      #   # good
+      #
+      #   class CreateUsers < ActiveRecord::Migration[7.0]
+      #   end
+      #
+      class MigrationFileName < Base
+        MSG = 'The class name does not match with the file name.'
+
+        def on_new_investigation
+          filepath = processed_source.file_path
+          @basename = File.basename(filepath, '.rb')
+        end
+
+        def on_class(node)
+          class_name = node.loc.name.source
+          snake_class_name = to_snakecase(class_name)
+          basename_without_timestamp = @basename.sub(/^[0-9]+_/, '')
+          add_offense(node.loc.name) if snake_class_name != basename_without_timestamp
+        end
+
+        private
+
+        def to_snakecase(word)
+          word.gsub!(/([A-Z\d]+)([A-Z][a-z])/, '\1_\2')
```

Comment:
I think this shouldn't mutate the word, i.e.:

```ruby
word
  .gsub(/([A-Z\d]+)([A-Z][a-z])/, '\1_\2')
  .gsub(/([a-z\d])([A-Z])/, '\1_\2')
  .tr('-', '_')
  .downcase
```

Reply:
That's true. Thanks!